Fix asset pipeline errors for plugin dummy apps

jonathanhefner opened the pull request:
To fix #43920, f292daad7051275f7b557b352876cfeb3988f639 added `sprockets-rails` to the generated `Gemfile` for engine plugins because their dummy apps use Sprockets.  However, non-engine plugins exhibit the same issue because their dummy apps also use Sprockets.

This commit forces `skip_asset_pipeline` to be true when a plugin is not an engine, and fixes several tests that failed to detect these issues because they were accidentally using the `rails/rails` `Gemfile` instead of the generated plugin `Gemfile`.

rafaelfranca reviewed (commented):
Can you split the refactoring from the commit that fix the issue?

jonathanhefner committed:
Fix asset pipeline errors for plugin dummy apps

To fix #43920, f292daad7051275f7b557b352876cfeb3988f639 added
`sprockets-rails` to the generated `Gemfile` for engine plugins because
their dummy apps use Sprockets.  However, non-engine plugins exhibit the
same issue because their dummy apps also use Sprockets.

This commit forces `skip_asset_pipeline` to be true when a plugin is not
an engine, and fixes several tests that failed to detect these issues
because they were accidentally using the `rails/rails` `Gemfile` instead
of the generated plugin `Gemfile`.
